update AWS credential setup docs, consolidate cloud auth docs

cg505 opened the pull request:
<!-- Describe the changes in this PR -->
Need to set up a redirect from `cloud-setup/cloud-auth.html` -> `cloud-setup/cloud-permissions/index.html`

Preview: https://docs.skypilot.co/en/svc-account-docs/docs/index.html

Fixes #4839.

<!-- Describe the tests ran -->
<!-- Unit tests (tests/test_*.py) are part of GitHub CI; below are tests that launch on the cloud. -->

<!-- CI commands (/-prefixed) can only be triggered by repo members -->

Michaelvll reviewed (commented):
Haven't finished the reading but here are some initial comments : )

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,195 @@
+
+2. In the **Permissions options** section, select "Attach existing policies directly". Depending on whether you want simplified or minimal permissions, follow the instructions below:
+
+    .. tab-set::
+
+        .. tab-item:: Simplified permissions
+
+            Search for the **AdministratorAccess** policy, and check the box to add it. Click **Next** to proceed.
+
+        .. tab-item:: Minimal permissions
+
+            Click on **Create Policy**.
+
+            .. image:: ../../images/screenshots/aws/aws-create-policy.png
+                :width: 80%
+                :align: center
+                :alt: AWS Create Policy
+
+            This will open a new window to define the minimal policy.
+
+            Choose "JSON" tab and copy the needed minimal policy rules.
+            **See** :ref:`aws-minimal-policy` **for all the policy rules.**
+
+            Click **Next: Tags** and follow the instructions to finish creating the policy. You can give the policy a descriptive name, such as ``minimal-skypilot-policy``.
+
+            Go back to the previous window and click on the refresh button, and you can now search for the policy you just created.
+
+            .. image:: ../../images/screenshots/aws/aws-add-policy.png
+                :width: 80%
+                :align: center
+                :alt: AWS Add Policy
+
+            Check the box to add the policy, and click **Next** to proceed.
```
The alternative is that this is outside the numbers, which looks worse IMO.
![image](https://github.com/user-attachments/assets/bf2ff352-fb97-4fac-b0d2-75b266580259)

I couldn't find a way to make it indented to the same level as the numbers, which would be ideal.

Michaelvll replied:
IIRC, if you have correct number of spaces for indent, it could show in a different way. If it does not work, I think the non-indent one is ok. The quote looks quite bad to me.

cg505 replied:
Got it. Need 3 spaces, not 2 or 4 :)

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,195 @@
+`AWS IAM Identity Center <https://aws.amazon.com/iam/identity-center/>`_ (Successor to AWS Single Sign-On, or SSO) is supported.
+
+.. warning::
+  SSO login *will not work across multiple clouds*. If you use multiple clouds, you should **LINK ME** :ref:`set up a dedicated user and credentials <dedicated-aws-user>` so that instances launched on other clouds can use AWS resources.
```
oh I fixed it but forgot to update the LINK ME, thanks

cg505 commented on docs/source/getting-started/installation.rst:
```diff
@@ -235,16 +235,19 @@ AWS
 
 
-To get the **AWS access key** required by :code:`aws configure`, please go to the `AWS IAM Management Console <https://us-east-1.console.aws.amazon.com/iam/home?region=us-east-1#/security_credentials>`_ and click on the "Access keys" dropdown (detailed instructions `here <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey>`__). The **Default reg ...
+To set up AWS credentials, log into the AWS console and `create an access key for yourself <https://docs.aws.amazon.com/IAM/latest/UserGuide/access-key-self-managed.html#Using_CreateAccessKey>`_. If you don't see the "Security credentials link", you may be using SSO - see :ref:`aws-sso`.
```
It's in AWS instructions - I'll clarify.

SeungjinYang commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,195 @@
+
+1. Open the `IAM dashboard <https://us-east-1.console.aws.amazon.com/iamv2/home#/home>`_ in the AWS console and click on the **Users** tab. Then, click **Create User** and enter the user's name. Click **Next**.
+
+TODO these screenshots are all outdated
```
Flagging this TODO comment

cg505 replied:
Thanks. Want to make sure this looks good in general then I'll redo all the screenshots at the end, since it's pretty rote.

SeungjinYang commented:
I finally got non-SSO AWS creds to work for me following this, so this doc is useful!
Take it or leave it, but I did have one hitch with the setup moving from SSO to long lived credentials specifically:
- I had `export AWS_PROFILE=AWSPowerUserAccess-xxxxxxxxxxx` set in my `~/.zshrc` already which I had to delete. Before I did that, `sky check aws` kept showing the SSO prompt (which makes sense in hindsight but it was confusing at the moment).
- `aws configure list` is a good sanity check after running `aws configure`, I initially had this
```
aws configure list
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile                <not set>             None    None
access_key                <not set>             None    None
secret_key                <not set>             None    None
    region                <not set>             None    None
```
and then when everything was in place, I had this
```
aws configure list
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile                <not set>             None    None
access_key     ****************xxxx shared-credentials-file    
secret_key     ****************xxxx shared-credentials-file    
    region                <not set>             None    None
```

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,195 @@
+Using AWS SSO
+-------------
+
+`AWS IAM Identity Center <https://aws.amazon.com/iam/identity-center/>`_ (Successor to AWS Single Sign-On, or SSO) is supported.
+
+.. warning::
+  SSO login *will not work across multiple clouds*. If you use multiple clouds, you should :ref:`set up a dedicated user and credentials <dedicated-aws-user>` so that instances launched on other clouds can use AWS resources.
+
+To use it, ensure that your machine `has AWS CLI V2 installed <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_ (by default, ``pip install skypilot[aws]`` installs V1; V2 cannot be installed via pip).
+
+You can use the following to check version:
+
+.. code-block:: console
+
+    $ aws --version
+    aws-cli/2.25.6 ...
+
+Visit your SSO login portal (e.g. ``https://my-sso-portal.awsapps.com/start``), and click on **Access keys** under the corresponding account. Under "AWS IAM Identity Center credentials (Recommended)" note:
+
+- the ``SSO start URL``
+- the ``SSO Region``
+
+Then, log into your SSO account:
+
+.. code-block:: console
+
+    $ aws configure sso
+
+- **SSO session name**: should be set, but you can choose any name you want.
+- **SSO start URL**: copy from the SSO login portal
+- **SSO region**: copy from the SSO login portal
+- **SSO registration scopes**: leave blank to use the default
+
+Log in and approve the request in your web browser. Then back in the CLI, complete the remaining fields:
+
+- **Default client Region**: optional
+- **CLI default output format**: optional
+- **Profile name**: set to ``default`` if you want to use this profile by default with SkyPilot, otherwise see :ref:`several-aws-profiles`.
+
+If everything is set up correctly, :code:`sky check aws` should succeed!
```
We need to emphasize the requirement of setting `AWS_PROFILE` in the env var. Otherwise, it won't work for SkyPilot.

cg505 replied:
If you follow the instructions to set profile name to `default` (line 46), it's not needed to set AWS_PROFILE. This part is already complicated so I don't want to add too much detail here. I'm adding a troubleshooting section that will include checking AWS_PROFILE.

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/gcp.rst:
```diff
@@ -1,19 +1,56 @@
 =============
 
 
+.. _gcp-service-account:
+
+GCP service account
+-------------------
+
+`GCP service accounts
+<https://cloud.google.com/iam/docs/service-account-overview>`__ are supported.
+
+.. tip::
+    A service account on your local machine can avoid the periodic
+    ``google.auth.exceptions.RefreshError: Reauthentication is needed. Please
+    run `gcloud auth application-default login` to reauthenticate.`` error. A
+    service account is long-lived as it does not have an expiry time.
+
+Set up a service account as follows:
+
+1. Follow the :ref:`instructions <gcp-service-account-creation>` to create a service account with the appropriate roles/permissions.
+2. In the "Service Accounts" tab in the `IAM & Admin console
+   <https://console.cloud.google.com/iam-admin/iam>`__, click on the service
+   account to go to its detailed page. Click on the **KEYS** tab, then click on
+   **ADD KEY** to add a JSON key.  The key will be downloaded automatically.
+3. Set the environment variable ``GOOGLE_APPLICATION_CREDENTIALS`` to the path of the key file, and configure the gcloud CLI tool:
+
+   .. code-block:: console
+
+       $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/key.json
+       $ gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
+       $ gcloud config set project your-project-id
+
+   You may want to add the export statement in your profile (e.g. ``~/.bashrc``, ``~/.zshrc``) so that it is set automatically in all new terminal sessions.
+
```
We should also check when/why we need `gcloud auth login` and `gcloud auth application-default login`. It is currently quite unclear in the doc and always confuse me.

cg505 replied:
To be clear, this is copied from the deleted file and I didn't change the content. But yes - you can just do `gcloud auth login --update-adc`. We should update this, but maybe not in this PR.

Michaelvll replied:
Sounds good. Just want to make sure this is on our radar. IIRC, the former is required for GCS access while the latter is needed for creating VMs.

Michaelvll replied:
Also, we should check if the service account credentials get uploaded correctly.

cg505 replied:
> Also, we should check if the service account credentials get uploaded correctly.

@Michaelvll Looks like it should: https://github.com/skypilot-org/skypilot/blob/e536958f1e9cd2bf91e379bd8381dbb8c084c0b1/sky/clouds/gcp.py#L898-L902 It will check the `GOOGLE_APPLICATION_CREDENTIALS` env var.

I will try it though to double check.

cg505 replied:
Uncovered #5210, but otherwise it works.

Michaelvll reviewed (commented):
Thanks @cg505! The changes look quite good to me from a high-level view. We may need to think of some alternatives of the title of `cloud-permissions/index.rst` page to make it easier to find for admins and usres.

cg505 commented:
> I finally got non-SSO AWS creds to work for me following this, so this doc is useful!

@SeungjinYang Thank you for the feedback! It's great to hear that this is helpful.

I added a "troubleshooting" section based on your issues.

Updated other things as per PR comments. Still need to fix the screenshots but otherwise ready for final passes.

SeungjinYang commented:
Content looks good to me! Just one minor thing that I can't figure out if it is just me, when I view https://docs.skypilot.co/en/svc-account-docs/cloud-setup/cloud-permissions/aws.html I sometimes see these yellow lines on some code block lines:
<img width="800" alt="Screenshot 2025-04-07 at 4 34 54 PM" src="https://github.com/user-attachments/assets/55986a50-2a85-4292-8809-7761ef08f752" />
<img width="794" alt="Screenshot 2025-04-07 at 4 36 15 PM" src="https://github.com/user-attachments/assets/3a656bd2-ce30-4b3e-9ee0-1521c3a8953a" />
I'm not sure where this is coming from since the code change doesn't do anything special for these lines :shrug:

cg505 commented:
> I'm not sure where this is coming from

@SeungjinYang There is an `:emphasize-lines:` argument on the code block. It's intentional.

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+               :align: center
+               :alt: AWS Add Policy
+
+           Check the box to add the policy, and click **Next** to proceed.
```
Can we move these steps into the `Create a user account with minimal permissions` below and just keep a link here to make it easier to read? Otherwise, it feels to long to scroll down.

cg505 replied:
I see what you mean.

> move these steps into the `Create a user account with minimal permissions`

That section is reduced significantly because most of the flow is the same for minimal permissions and for simple permissions. It doesn't make sense to move everything into the "Minimal permissions" section because these only apply to user creation, and "Minimal permissions" could apply to both user creation and role creation.

That said, I think this can be better. Will do another pass

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+Setting up a dedicated SkyPilot IAM user
+----------------------------------------
+
+You can create a dedicated IAM user for SkyPilot, if you can't or don't want to use your normal credentials, or if you want to restrict SkyPilot's permissions.
```
We should emphasize here that a dedicated user is not needed if `sky check aws` works, and we can put the reason for creating a dedicated IAM user in a itemized list.

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+
+.. _dedicated-aws-user:
+
+Setting up a dedicated SkyPilot IAM user
```
How about using a shorter title, so as to make sure the right TOC does not wrap? Otherwise, it looks a bit hard to see the structure of the page in that TOC.
```suggestion
Dedicated SkyPilot IAM user
```

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+
+.. _several-aws-profiles:
+
+Using several profiles or accounts
```
Same as above: make the TOC title short
```suggestion
Switch profiles or accounts
```

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+
+.. _aws-troubleshooting:
+
+Troubleshooting issues
```
```suggestion
Troubleshooting
```

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+       secret_key     ****************xxxx              sso
+           region                <not set>             None    None
+
+2. **Check** ``sky check aws``. This should show whether SkyPilot is picking up the credentials and has the necessary permissions.
+
+   .. code-block:: console
+
+       $ sky check aws
+       Checking credentials to enable clouds for SkyPilot.
+         AWS: enabled [compute, storage]
+       ...
```
How about using `sky check aws -v` as the first step? It will show what credential is being used and also see if things has been setup.

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+         AWS: enabled [compute, storage]
+       ...
+
+**Common issues:**
```
```suggestion
Common issues
~~~~~~~~~~~~~~~~~
```

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+
+      # .bashrc / .zshrc
+      export AWS_PROFILE='AWSPowerUserAccess-xxxxxxx'
+
```
We may want to add an API server item here for switching account issue with a remote API server. As discussed offline, we can fix the local API server for taking in `env_vars` starts with `AWS_`

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/gcp.rst:
```diff
@@ -1,19 +1,56 @@
 =============
 
 
+.. _gcp-service-account:
```
same as the comment above, we should have a sentence at the begining referring to normal installation and setup, and claiming that the following is optional

Michaelvll commented on docs/source/examples/managed-jobs.rst:
```diff
@@ -625,6 +625,28 @@ you can still tear it down manually with
+
+- **AWS**: :ref:`Create a dedicated SkyPilot IAM user <dedicated-aws-user>`.
+- **GCP**: :ref:`Create a GCP service account <gcp-service-account>`.
+- **Other clouds**: Make sure you are using credentials that do not expire.
```
We should have kubernetes' guide linked here as well?

cg505 replied:
We don't have a good guide for Kubernetes. I think there's no consistent way to do this since authentication may vary for different deployments. The existing "service account" guide is not relevant - service accounts only apply intra-cluster.

cg505 replied:
> I think there's no consistent way to do this

This is wrong. We can create static credentials for k8s service accounts. However, we don't have good docs on how to do this except for the main setup docs, which include things like creating namespaces (not relevant). I need to follow up on this.

Michaelvll commented on docs/source/cloud-setup/cloud-permissions/gcp.rst:
```diff
@@ -1,19 +1,56 @@
+
+.. _cloud-permissions-gcp:
+
+Setting permissions
```
Since we have service account at the top already, should rename the `Service account` below to `Service account for SkyPilot system`. Also, in AWS page, we may be able to rename the title of IAM role to be `IAM role for SkyPilot system` as well

Michaelvll reviewed (commented):
Thanks @cg505! Left some comments.

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+
+`AWS IAM Identity Center <https://aws.amazon.com/iam/identity-center/>`_ (Successor to AWS Single Sign-On, or SSO) is supported.
+
+.. warning::
+  SSO login *will not work across multiple clouds*. If you use multiple clouds, you should :ref:`set up a dedicated user and credentials <dedicated-aws-user>` so that instances launched on other clouds can use AWS resources.
```
It can break cross-cloud s3 access as well - e.g. user creates a sky-managed named bucket, and it is created on AWS. Then later a GCP instance will not be able to access that bucket.

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+
+.. code-block:: console
+
+    $ aws --version
+    aws-cli/2.25.6 ...
```
Took a stab at adding a potential solution. It's not a great situation though. TBH would be nice if we just avoided depending on aws-cli directly.

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+   .. image:: ../../images/screenshots/aws/aws-add-user.png
+       :width: 80%
+       :align: center
+       :alt: AWS Add User
```
I did experiment with it, but the AWS console is really tall... If we make the images less wide then the text gets really small. I could vertically crop some of them but not all. Tried to keep them the same size, but maybe better to crop each one as much as possible. What do you think?

Michaelvll reviewed (approved):
Thanks @cg505 for making the update! The latest doc mostly looks good to me! cc'ing @concretevitamin if you want to do a final check.

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,20 +1,43 @@
+  .. [1] To allow managed jobs to run on AWS instances, make sure your controller is also on AWS, by :ref:`specifying the controller resources <jobs-controller-custom-resources>`.
+
+
+To use SSO, ensure that your machine `has AWS CLI v2 installed <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_. By default, ``pip install skypilot[aws]`` installs v1; v2 cannot be installed via pip. If you install v2, you may need to ``deactivate`` (for venv) or ``conda deactivate`` to avoid the v1 CLI taking precedence.
```
I think this path (/usr/local/aws-cli/) is MacOS-specific, but let me double check.

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -4,36 +4,330 @@
+
+    $ # A cluster launched under a different profile.
+    $ AWS_PROFILE=AdministratorAccess-12345 sky launch --cloud aws -c my-sso-cluster
+
```
#5229

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,384 @@
+6. Use the newly created access key to configure your credentials with the AWS CLI:
+
+   .. code-block:: console
+     :emphasize-lines: 9-10
```
updated

cg505 commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,384 @@
+
+       .. tab-item:: Minimal permissions
+
+           Click on **Create Policy**.
+
+           .. image:: ../../images/screenshots/aws/aws-create-policy.png
+               :alt: AWS Create Policy
+
+           This will open a new window to define the minimal policy. Follow the instructions to create a new policy: :ref:`aws-minimal-policy`.
+
+           Come back to this window, and in the **Permissions policies** box, click on the refresh button. You can now search for the policy you just created.
+
+           .. image:: ../../images/screenshots/aws/aws-add-policy.png
+               :alt: AWS Add Policy
+
+           Check the box to add the policy, and click **Next** to proceed.
```
Since the instructions here are specific to creating a user, I think we should keep them here. I moved down more of the policy-specific instructions.

cg505 commented:
@concretevitamin This is basically ready to go, do you want to do a pass?

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/index.rst:
```diff
@@ -1,15 +1,24 @@
+==============================
 
-.. note::
+Individual clouds may have unique advanced configuration options. These are optional and mostly targeted at cloud admins. **For most users, SkyPilot should work out of the box** with the permissions you have configured locally during :ref:`installation <cloud-account-setup>`.
```
```suggestion
Some providers in SkyPilot may offer identity and permission-related configuration options. These are **optional** and mostly targeted at cloud admins. **For most users, SkyPilot should work out of the box** with the permissions you have configured locally during :ref:`installation <cloud-account-setup>`.
```

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,385 @@
+
+.. _several-aws-profiles:
+
+Switch profiles or accounts
```
Pull this up to 2nd subsection? Since this one is lower-lift, potentially user-facing.

How about:
- Using AWS SSO
- Using a specific VPC
- Switching profiles or accounts
- <the rest, admin-facing>

cg505 replied:
Hmm, I feel that "Using a specific VPC" is probably more admin-facing than "Dedicated SkyPilot IAM user". Anyway, I'm pretty happy with any order as long as SSO is at the top.

cg505 replied:
Reordered:

- Using AWS SSO
  - Multi-cloud access with SSO login
- Switching profiles or accounts
- Using a specific VPC
- Dedicated SkyPilot IAM user
- Minimal permissions
  - Create a user account with minimal permissions
  - Create the internal IAM role for SkyPilot
  - Minimal IAM policy
- Troubleshooting
  - Common issues

@concretevitamin what do you think?

concretevitamin replied:
Sounds good.

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,385 @@
+Using AWS SSO
+-------------
+
+`AWS IAM Identity Center <https://aws.amazon.com/iam/identity-center/>`_ (Successor to AWS Single Sign-On, or SSO) is supported.
```
```suggestion
`AWS IAM Identity Center <https://aws.amazon.com/iam/identity-center/>`_ (successor to AWS Single Sign-On, or SSO) is supported.
```

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,385 @@
+
+`AWS IAM Identity Center <https://aws.amazon.com/iam/identity-center/>`_ (Successor to AWS Single Sign-On, or SSO) is supported.
+
+.. _sso-feature-compat:
+
+SSO login has limited functionality *across multiple clouds*. If you use multiple clouds, you can :ref:`set up a dedicated IAM user and access key <dedicated-aws-user>` so that instances launched on other clouds can use AWS resources.
+
+.. list-table::
+   :header-rows: 1
+
+   * - *Supported features:*
+     - SSO credentials
+     - Static credentials
+   * - Use S3 buckets on an AWS cluster
+     - |:white_check_mark:|
+     - |:white_check_mark:|
+   * - Use S3 buckets on a cluster in another cloud
+     - |:x:|
+     - |:white_check_mark:|
+   * - Run :ref:`managed jobs <managed-jobs>` across multiple clouds
+     - |:yellow_circle:| [1]_
+     - |:white_check_mark:|
+
+.. [1] To allow managed jobs to run on AWS instances, make sure your controller is also on AWS, by :ref:`specifying the controller resources <jobs-controller-custom-resources>`.
+
```
These are multicloud-related notes. Can we move it down, behind the instructions below? Can even be a subsection, "SSO and Cross-cloud Features"?

cg505 replied:
I think it's pretty important for SSO users to see that some features won't work, otherwise it will just seem like SkyPilot is broken. We could leave a very short note here and link to a subsection with the details below, though.

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,385 @@
+    $ # A cluster launched under a different profile.
+    $ AWS_PROFILE=AdministratorAccess-12345 sky launch --cloud aws -c other-profile-cluster
+
+If you are using a :ref:`remote API server <sky-api-server>`, the AWS credentials are configured on the remote server. Overriding ``AWS_PROFILE`` won't work.
```
```suggestion
If you are using a :ref:`remote API server <sky-api-server>`, the AWS credentials are configured on the remote server. Overriding ``AWS_PROFILE`` on the client side won't work.
```

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,385 @@
+       $ sky check aws -v
+       Checking credentials to enable clouds for SkyPilot.
+         AWS: enabled [compute, storage]
+           Activated account: VRSC9IFFYQI7THCKR5UVC [account=190763068689]
```
Is this random string already?

cg505 replied:
yes

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,385 @@
+Common issues
+~~~~~~~~~~~~~
+
+- **Wrong profile is enabled.** SkyPilot will respect the ``AWS_PROFILE`` environment variable if it is set - see :ref:`several-aws-profiles`. If ``AWS_PROFILE`` is not set, SkyPilot will use the profile named ``default``.
```
```suggestion
- **Wrong profile is enabled.** SkyPilot will respect the ``AWS_PROFILE`` environment variable if it is set; see :ref:`several-aws-profiles`. If ``AWS_PROFILE`` is not set, SkyPilot will use the profile named ``default``.
```

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -1,39 +1,385 @@
+
+.. _several-aws-profiles:
+
+Switch profiles or accounts
```
```suggestion
Switching profiles or accounts
```

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
Nit: nested dirs. Can refactor in the future.

concretevitamin commented on docs/source/getting-started/installation.rst:
```diff
@@ -235,19 +235,22 @@ AWS
 
 
-To get the **AWS access key** required by :code:`aws configure`, please go to the `AWS IAM Management Console <https://us-east-1.console.aws.amazon.com/iam/home?region=us-east-1#/security_credentials>`_ and click on the "Access keys" dropdown (detailed instructions `here <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey>`__). The **Default reg ...
+To set up AWS credentials, log into the AWS console and `create an access key for yourself <https://docs.aws.amazon.com/IAM/latest/UserGuide/access-key-self-managed.html#Using_CreateAccessKey>`_. If you don't see the "Security credentials" link shown in the AWS instructions, you may be using SSO - see :ref:`aws-sso`.
```
```suggestion
To set up AWS credentials, log into the AWS console and `create an access key for yourself <https://docs.aws.amazon.com/IAM/latest/UserGuide/access-key-self-managed.html#Using_CreateAccessKey>`_. If you don't see the "Security credentials" link shown in the AWS instructions, you may be using SSO; see :ref:`aws-sso`.
```

concretevitamin commented on docs/source/getting-started/installation.rst:
```diff
@@ -235,19 +235,22 @@ AWS
   # Configure your AWS credentials
   aws configure
 
+- For **AWS Access Key ID**, copy the "Access key" value from console
```
```suggestion
- For **AWS Access Key ID**, copy the "Access key" value from console.
```

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/index.rst:
```diff
@@ -1,15 +1,24 @@
 
-Minimal Cloud Permissions
-=================================
+Cloud Accounts and Permissions
```
Another idea: "Cloud Identities and Permissions". No strong preference.

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/gcp.rst:
```diff
@@ -1,19 +1,60 @@
 
+.. note::
+
+    By default, SkyPilot will pick up the credentials you have set up. For most cases, the :ref:`getting started instructions <installation-gcp>` are all you need to do. The steps below are **optional advanced configuration options**, aimed primarily at cloud admins and advanced users.
```
```suggestion
    By default, SkyPilot will pick up the credentials you have set up. For most cases, the :ref:`installation instructions <installation-gcp>` are all you need to do. The steps below are **optional advanced configuration options**, aimed primarily at cloud admins and advanced users.
```
Same for another page?

cg505 replied:
need to fix other page

concretevitamin commented on docs/source/examples/managed-jobs.rst:
```diff
@@ -625,6 +625,28 @@ you can still tear it down manually with
+
+Since the :ref:`jobs controller <jobs-controller>` is a long-lived instance that will manage other cloud instances, it's best to **use static credentials that do not expire**. If a credential expires, it could leave the controller with no way to clean up a job, leading to expensive cloud instance leaks.
+
+To use static credentials for the jobs controller, just make sure the static credentials are in use by SkyPilot. They will be automatically uploaded to the jobs controller. **If you're already using local credentials that don't expire, no action is needed!**
```
```suggestion
To use static credentials for the jobs controller, just make sure the static credentials are in use by SkyPilot. They will be automatically uploaded to the jobs controller. **If you're already using local credentials that don't expire, no action is needed.**
```

concretevitamin commented on docs/source/examples/managed-jobs.rst:
```diff
@@ -625,6 +625,28 @@ you can still tear it down manually with
+
+.. _static-creds:
+
+Using static credentials
```
Q: is "static credentials" the right term? I've always associated with a static file, e.g., ~/.aws/credentials. But what we referred to below = service accounts?

Do both static-file and service-accounts work here?

cg505 replied:
Here's my intended meaning:
- "static credentials" - a credential file that does not need to be refreshed/expire, like ~/.aws/credentials
- "service account" - the user role/account used by a service. In GCP, you can create a static credential that authenticates a service account. In AWS, there's no exact service account concept - you can create instance profiles to assume roles, or create IAM users with attached roles which could use static credentials.

Basically "service account" is just kind of an overloaded term. In GCP, the thing you have to create is literally called "service account" in the console, so that's why it is used below.

concretevitamin replied:
Maybe add the specific AWS and GCP examples as a short sentence here will help? E.g.,  ~/.aws/credentials on AWS or a GCP service account that uses GOOGLE_xxx_env_var=yyy.json?

concretevitamin replied:
Still thinking "Using static credentials" is potentially confusing. Wdyt about "Using long-lived credentials"?

cg505 replied:
Do you think "Using long-lived static credentials" is too much?

concretevitamin replied:
Reason I wanted to avoid "static credentials" is because it's not a generic term across clouds. "long-lived" or "non-expirable" is more generic.

cg505 replied:
Do you think "Using long-lived static credentials" is too much?

cg505 replied:
Do you think "Using long-lived static credentials" is too much?

concretevitamin commented on docs/source/examples/managed-jobs.rst:
```diff
@@ -706,7 +728,7 @@ Best practices for scaling up the jobs controller
 
 .. tip::
-  For managed jobs, it's highly recommended to use service accounts for :ref:`cloud authentication <cloud-auth>`. This is so that the jobs controller credentials do not expire. This is particularly important in large production runs to avoid leaking resources.
+  For managed jobs, it's highly recommended to use :ref:`static credentials for cloud authentication <static-creds>`. This is so that the jobs controller credentials do not expire. This is particularly important in large production runs to avoid leaking resources.
```
see above

concretevitamin reviewed (commented):
Awesome revamp @cg505. Left some comments.

concretevitamin commented on docs/source/cloud-setup/cloud-permissions/aws.rst:
```diff
@@ -3,18 +3,60 @@ AWS
+
+.. note::
+
+    If you want to use multiple clouds with SkyPilot, check the :ref:`SSO multi-cloud compatibility notes <sso-feature-compat>`.
```
```suggestion
    If you use AWS SSO and multiple clouds, check the :ref:`SSO multi-cloud compatibility notes <sso-feature-compat>`.
```

concretevitamin reviewed (approved):
LGTM, thanks @cg505. A final naming suggestion, up to you.
